cozy/codegen/cxx.py

```diff
@@ -272,7 +272,7 @@ def construct_concrete(self, t : Type, e : Exp, out : Exp):
                 self.construct_map(t, e, out))
         elif isinstance(t, THandle):
             return SEscape("{indent}{lhs} = {rhs};\n", ["lhs", "rhs"], [out, e])
-        elif is_scalar(t):
+        elif is_scalar(t) or isinstance(t, TTuple):
```

Comment:
Let's make this change in a different PR. It is likely more subtle than this:
 - In C++, tuples are stack-allocated value types that make copies when you use the `=` assignment operator
 - In Java, tuples are heap-allocated reference types that make aliases when you use the `=` assignment operator

Cozy *does* generate code that says to modify a tuple in-place, so for Java we need to be careful to copy the object instead of reference it.